ui/wobbly.js

```diff
@@ -307,18 +307,34 @@ function enableWobblyFx(wm) {
         SETTINGS_HANDLER = Settings.connect('changed::wobbly-effect', actionWobblyEffectSetting);
         actionWobblyEffectSetting(Settings, 'wobbly-effect');
     });
+
+    return GLib.SOURCE_REMOVE;
+}
+
+function enableWobblyFx(wm) {
+    initAnimationsDbus(wm);
 }
 
 function disableWobblyFx(wm) {
     Settings.disconnect(SETTINGS_HANDLER);
 
     getAnimatableWindowActors().forEach(actor => {
-        actor._animatableSurface = null;
+        if (actor._animatableSurface) {
+            wm._animationsServer.unregister_surface(actor._animatableSurface);
```

Comment:
Do we still need to call `unregister_surface` here manually given this is done when invoking `animationsServer.stop()` below?

Reply:
Can’t hurt. I think it’s best for each bit of code to clean up the strong references it holds to resources.